docs/proposals/kubelet-raw-metrics-api.md

```diff
@@ -0,0 +1,120 @@
+@<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/compute-resource-metrics-api.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Kubelet raw metrics API
+
+Kubelet currently exposes raw container metrics through the `/stats/` endpoint
+that serves raw container stats. However, this endpoint serves individual
+container stats, and applications like heapster, which aggregates metrics across
+the cluster, must repeatedly query for each container. The high QPS combined
+with the potential data size of raw stats puts unnecessary load on the system
+that could be avoided with an aggregate API.
+
+## Requirements:
+- Provide machine level metrics, all pod metrics (in single request), specific
+  pod metrics
+- Ability to authenticate machine & pod metrics independently from each other
+- Eventually support multiple kinds of metrics (e.g. derived/aggregate type)
+- Follow existing API conventions, fully compatible types able to eventually be
+  served by apiserver library
+
+## Design
+
+Kubelet will start providing a versioned API, following convensions of the API
+server. The metrics API will be its own [API group](api-group.md), and may
+eventually be reused at the cluster-level (outside the scope of this proposal).
+
+### Endpoints
+
+All endpoints are GET endpoints, rooted at `/apis/metrics/v1alpha1/`
```

Comment:
Addressed below, but I think it should be a separate endpoint, unless we can fit them neatly into the same metrics data type.